taipy/gui/data/pandas_data_accessor.py

```diff
@@ -267,18 +267,47 @@ def __get_data(  # noqa: C901
                 col = fd.get("col")
                 val = fd.get("value")
                 action = fd.get("action")
-                if isinstance(val, str):
-                    if self.__is_date_column(t.cast(pd.DataFrame, df), col):
-                        val = datetime.fromisoformat(val[:-1])
+                match_case = fd.get("matchCase", False)
+
+                # Only test if the column is a date column if val is a string
+                if isinstance(val, str) and self.__is_date_column(t.cast(pd.DataFrame, df), col):
```

Comment:
why do you change the previous code that was working ?
We want to build a query and apply it once.
the previous code
```py
                if isinstance(val, str):
                    if self.__is_date_column(t.cast(pd.DataFrame, df), col):
                        val = datetime.fromisoformat(val[:-1])
                    vars.append(val)
                val = f"@vars[{len(vars) - 1}]" if isinstance(val, (str, datetime)) else val
                right = f".str.contains({val})" if action == "contains" else f" {action} {val}"

```

The new code should be 
```py
                match_case = fd.get("matchCase", False) is not False
                right = None
                if isinstance(val, str):
                    if self.__is_date_column(t.cast(pd.DataFrame, df), col):
                        val = datetime.fromisoformat(val[:-1])
                    elif not match_case:
                        if action != "contains":
                            col = f"`{col}`.str.lower()"
                        val = val.lower()
                    vars.append(val)
                    val = f"@vars[{len(vars) - 1}]"
                    if action == "contains":
                        right = f".str.contains({val}{'' if match_case else ', case=False'})"
                if right is None:
                    right = f" {action} {val}"
```

Reply:
Apologies for such mistakes, I have made the requested changes.
Please check and review it. Let me if further changes are needed.

Reply:
Also all the existing tests in `test_pandas_data_accessor.py` are passing after the changes made.